src/local_manifest.rs

```diff
@@ -0,0 +1,1544 @@
+        match op {
+            CompareOp::Eq => self == other,
+            CompareOp::Ne => self != other,
+            _ => unimplemented!(),
+        }
+    }
+
+    fn get_chunks<'p>(&self, py: Python<'p>, block: usize) -> PyResult<&'p PyTuple> {
+        let elements = self
+            .0
+            .get_chunks(block)
+            .cloned()
+            .unwrap_or_default()
+            .into_iter()
+            .map(|x| Chunk(x).into_py(py));
+        Ok(PyTuple::new(py, elements))
+    }
+
+    fn is_reshaped(&self) -> PyResult<bool> {
+        Ok(self.0.is_reshaped())
+    }
+
+    fn assert_integrity(&self) -> PyResult<()> {
+        let result = panic::catch_unwind(|| self.0.assert_integrity());
+        match result {
+            Ok(_) => Ok(()),
+            Err(_) => Err(PyAssertionError::new_err("assert")),
+        }
+    }
+
+    #[classmethod]
+    fn from_remote(_cls: &PyType, remote: FileManifest) -> PyResult<Self> {
+        Ok(Self(
+            libparsec::client_types::LocalFileManifest::from_remote(remote.0)
+                .map_err(PyValueError::new_err)?,
+        ))
+    }
+
+    fn to_remote(&self, author: DeviceID, timestamp: DateTime) -> PyResult<FileManifest> {
+        Ok(FileManifest(
+            self.0
+                .to_remote(author.0, timestamp.0)
+                .map_err(PyValueError::new_err)?,
+        ))
+    }
+
+    #[classmethod]
+    // Python signature includes variables that are unused in LocalFileManifest
+    #[allow(unused_variables)]
+    fn from_remote_with_local_context(
+        _cls: &PyType,
+        remote: FileManifest,
+        prevent_sync_pattern: &PyAny,
+        local_manifest: Self,
+        timestamp: &PyAny,
+    ) -> PyResult<Self> {
+        Self::from_remote(_cls, remote)
+    }
+
+    fn match_remote(&self, remote_manifest: &FileManifest) -> PyResult<bool> {
+        Ok(self.0.match_remote(&remote_manifest.0))
+    }
+
+    /*
+     * BaseLocalManifest
+     */
+
+    #[getter]
+    fn base_version(&self) -> PyResult<u32> {
+        Ok(self.0.base.version)
+    }
+
+    #[getter]
+    fn id(&self) -> PyResult<EntryID> {
+        Ok(EntryID(self.0.base.id))
+    }
+
+    #[getter]
+    fn created(&self) -> PyResult<DateTime> {
+        Ok(DateTime(self.0.base.created))
+    }
+
+    #[getter]
+    fn is_placeholder(&self) -> PyResult<bool> {
+        Ok(self.0.base.version == 0)
+    }
+
+    fn dump_and_encrypt<'p>(&self, py: Python<'p>, key: &SecretKey) -> PyResult<&'p PyBytes> {
+        Ok(PyBytes::new(py, &self.0.dump_and_encrypt(&key.0)))
+    }
+
+    #[classmethod]
+    fn decrypt_and_load(_cls: &PyType, encrypted: &[u8], key: &SecretKey) -> PyResult<Self> {
+        match libparsec::client_types::LocalFileManifest::decrypt_and_load(encrypted, &key.0) {
+            Ok(x) => Ok(Self(x)),
+            Err(err) => Err(PyValueError::new_err(err)),
+        }
+    }
+
+    #[classmethod]
+    #[args(blocksize = "512 * 1024")]
```

Comment:
Should use `libparsec::api_types::DEFAULT_BLOCK_SIZE` instead

Reply:
It should but, `args` require string, and the constant `DEFAULT_BLOCK_SIZE` isn't a string.

Reply:
then keep the default at `None`, then use the constant in the function body

Reply:
done